### PR TITLE
97 remove password field from update user request and create endpoint for reset password

### PR DESCRIPTION
**Pull request by jVucetic**

### CODE REVIEW

File: src/main/java/com/g1dra/foodmanager/dto/users/CreateUserRequest.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-public class UserRequest {
+public class CreateUserRequest {
```
Comment (LjSamardzic) [inline src/main/java/com/g1dra/foodmanager/dto/users/CreateUserRequest.java:21]: We no longer need the @UpdateRolePermission here, right?

File: src/main/java/com/g1dra/foodmanager/dto/users/UpdateUserRequest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class UpdateUserRequest {
+    @NotEmpty(message = "Name is required")
+    public String name;
```
Comment (LjSamardzic) [inline src/main/java/com/g1dra/foodmanager/dto/users/UpdateUserRequest.java:18]: Is there any reason for some variables to be public? In the CreateUserRequest as well

File: src/main/java/com/g1dra/foodmanager/factory/UserFactory.java [hunk 3]
```diff
@@ -40,25 +41,24 @@ public static User createUser(UserRole role) {
         );
     }
 
-    public static UserRequest createRegularUserRequest()
+    public static CreateUserRequest createUserRequest(UserRole role)
```
Comment (LjSamardzic) [inline src/main/java/com/g1dra/foodmanager/factory/UserFactory.java:44]: It's great you added these!

**Review by LjSamardzic — CHANGES_REQUESTED** [review]

File: src/main/java/com/g1dra/foodmanager/dto/users/UpdateUserRequest.java (hunk 2 above)
Comment (jVucetic) [reply src/main/java/com/g1dra/foodmanager/dto/users/UpdateUserRequest.java]: No, that should be private.
I changed them to be private in CreateUserRequest, UpdateUserRequest and in User model as well

**Pull request merged**